### DISCUSSION TITLE
What should I do to make SD3 work on Foocus??

### DISCUSSION THREAD

**Discussion by oldhand7** [lllyasviel/Fooocus#3145] [Q&A] [answered]
I would love to implement this model in Foocus.

**Comment by southbayjay** [comment]
That's the million dollar question we all want to know..

**Comment by IPv6** [comment, marked as answer]
To make SD3 work in Fooocus the following needed (afaik)

- initially this PR should be completed https://github.com/lllyasviel/Fooocus/pull/3084 
- then prompt encoding via t5 etc etc should be implemented - with VAE tiling and all the stuff

- figure out loras, controlnet and inpainting for SD3. Or wait while it will be implemented in ComfyUI/etc to look at compatible approach. Currently there is none.

- train from scratch controlnets models and iron out conditioning (without them Fooocus is not Fooocus, imho). There is a separate issue of faceswap - IPAdapter+appendix tech implementation has specifics. Controlnets can be delayed until ComfyUI will have them. Currently there is none. 
 
- train from scratch inpainting model. Inpainting is fooocus-specific (  @lllyasviel  explained details somewhere ). Also can be delayed/skipped (SDXL can be used for that)

- retrain GPT2 prompt expander, rewrite styles positive/negative prompts, replace image describer to support SD3 prompting best practices, etc. seems prompting is somewhat different

- Finally iron out UI/UX. Resolutions, image2image integration in pipelines, sort out incompatibilities somehow (freeu, hyper-sd, lighting)

**Reply by mashb1t** [reply]
@IPv6 totally agree. Thanks for the summary, much appreciated!

**Reply by xhoxye** [reply]
The work needs to be done step by step, and it cannot be done all at once

**Comment by mashb1t** [comment]
@oldhand7 you've already created [this discussion](https://github.com/lllyasviel/Fooocus/discussions/3142) yesterday. Have you familiarized yourself with the codebase since https://github.com/lllyasviel/Fooocus/discussions/2302 / https://github.com/lllyasviel/Fooocus/discussions/2292?

**Comment by IPv6** [comment]
Well, community started to fill the gaps :) ControlNets are already on the way: https://huggingface.co/InstantX/SD3-Controlnet-Canny_alpha_512/blob/main/README.md

SD3 is now also supported in https://github.com/runew0lf/RuinedFooocus - via Comfy UI backend

**Reply by nikolatesla20** [reply]
yes but is the inpainting good? The insanely powerful inpaint is why Fooocus rocks

**Reply by IPv6** [reply]
> yes but is the inpainting good? The insanely powerful inpaint is why Fooocus rocks

currently there are no dedicated SD3 inpainting model and workflow - at least i did not see anything alike, not in RuinedFooocus, not in Comfy. It should be possible, but nothing publicity published, afaik
